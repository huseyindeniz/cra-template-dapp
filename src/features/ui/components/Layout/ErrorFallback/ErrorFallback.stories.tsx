// ConnectButton.stories.ts|tsx
import type { Meta, StoryObj } from '@storybook/react';

import { ErrorFallback } from './ErrorFallback';

const meta: Meta<typeof ErrorFallback> = { component: ErrorFallback };
export default meta;

type Story = StoryObj<typeof ErrorFallback>;

export const Default: Story = {
  args: {
    error: new Error('Mock error message'),
  },
};
