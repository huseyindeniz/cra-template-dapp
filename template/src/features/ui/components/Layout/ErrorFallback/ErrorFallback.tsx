import { Box } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';

import { AlertMessage } from '../../AlertMessage/AlertMessage';

export interface ErrorFallbackProps {
  error: Error;
}

export const ErrorFallback: React.FC<ErrorFallbackProps> = ({ error }) => {
  const { t } = useTranslation('Layout');
  return (
    <Box m={10}>
      <AlertMessage status="error" title={t('An error occured!')}>
        {error.message}
      </AlertMessage>
    </Box>
  );
};
