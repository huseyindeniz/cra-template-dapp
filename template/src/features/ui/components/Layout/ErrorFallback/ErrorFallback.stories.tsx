// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { ErrorFallback } from './ErrorFallback';

export default {
  title: 'ui/Components/Layout/ErrorFallback',
  component: ErrorFallback,
} as ComponentMeta<typeof ErrorFallback>;

export const Default: ComponentStory<typeof ErrorFallback> = args => (
  <ErrorFallback error={new Error('Mock error message')} />
);
