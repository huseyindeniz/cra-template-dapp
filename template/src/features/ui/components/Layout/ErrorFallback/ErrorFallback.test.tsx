import { render } from '@testing-library/react';

import { ErrorFallback } from './ErrorFallback';

describe('Feature: UI', () => {
  describe('Component: Layout/ErrorFallback', () => {
    it('should be visible and show error message', () => {
      // Arrange
      const error = new Error('Mock error message');
      const { asFragment, getByText } = render(<ErrorFallback error={error} />);
      // Assert
      expect(getByText('An error occured!')).toBeVisible();
      expect(getByText('Mock error message')).toBeVisible();
      expect(asFragment).toMatchSnapshot();
    });
  });
});
