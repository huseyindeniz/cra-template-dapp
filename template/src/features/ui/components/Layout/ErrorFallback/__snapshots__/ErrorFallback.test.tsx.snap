// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: UI Component: Layout/ErrorFallback should be visible and show error message 1`] = `[Function]`;
